src/CombinePDF/CombinePDF.tsx

```diff
@@ -23,6 +23,23 @@ async function mergePdfs(files: File[], addToDoc?: PDFDocument) {
   };
 }
 
+// This is a work around for the pdf-js library because removePage method does not delete all
```

Comment:
@mkhatib i'm kinda lost here, according to https://github.com/Hopding/pdf-lib/issues/140#issuecomment-515140638 

the maintainer mentions that ```removePage``` ONLY delete a reference of the page from the pdf leaving objects and size same. I went with the solution he mentioned on deleting all of the page reference objects etc .. assuming the user want to delete the whole page and reduce the size ? 

Also i didn't test this throughly so not so sure on what to do here.

What's your opinion ?

Reply:
I think a better way might be to handle this similar to how we handle Ordering Pages. We do not recreate or reorder the actual PDF pages instead we just maintain a pages order list which is the indexes of the PDF pages and how they should be EVENTUALLY ordered when the user downloads the PDF.

I think the same can be achieved for deletion. You wouldn't touch the PDF file at all when deleting. Instead you just delete the "page index" from the `pagesOrder` list - react will re-render the list and only show the user the pages that exist in the `pagesOrder` - so the user will have instant feedback. 

When the user is ready to click DOWNLOAD. We already only download the pages in that `ordersPage` so anything else won't be included. I think this is the right way to go about this - let me know if I am missing anything.

Although in order to address the delete issue you might need to do a simple modification on `getOrderedPdf` to use the order list instead of getting all the pages. 

Let me know if this makes senes.

Reply:
I think this is defiantly cleaner

Reply:
Actually one edge case i'm thinking for this approach is.

If the user wants to upload a new document and append to existing list, currently when a file is dropped ```mergePdfs``` gets called re-creating all of the documents list (including the ones we deleted, since it's already referenced in original document and we only mutated the order page array) also a re-render happens to render the old, new, deleted pages. 

So i'm thinking we might need to still remove the page from document somehow.

If it doesn't make sense i can elaborate more ?

Reply:
I think you're right this will introduce an issue but not exactly what you described, I think it's solvable but let me describe what you're saying and what could happen in an example.
| action        | actual pages in pdf | pages order | 
| ------------- | ------------- | ------------- | 
| upload 3 pages  | 1 2 3  | 1 2 3  | 
| reorders to 3 2 1  | 1 2 3  | 3 2 1  | 
| uploads 2 pages |   1 2 3 4 5  | 3 2 1 4 5  | 
| delete index 2 |   1 2 3 4 5  | 3 1 4 5  | 
| upload 2 pages |   1 2 3 4 5 6 7  | 3 1 4 5 5 6  | 

The issue when a user uploads after a deletion (look at the repeated index in the last row). And the fix should be simple (I think). All is needed is to fix our reference of "page count" when appending the the document to reference the actual PDF document w're handling, instead of what we're doing now which is using `pagesOrder` list as references to what are the indexes of newly uploaded pages should be.

This is the bit of code needs changing (code change is between **):

```
      // Calculate new page order with the added pages at the end of the order.
      // To avoid losing previously ordered indexes.
     **const oldPdfDoc = doc;
     const newPagesIndexOffset = oldPdfDoc.getPageCount();**

      const oldPagesOrder = pagesOrder; 
      const oldPageCount = oldPagesOrder.length;
      const newlyAddedPagesOrder = new Array(pageCount - oldPageCount)
        .fill(0)
        .map((_, index) => **newPagesIndexOffset** + index + 1);
      const newOrder = [...oldPagesOrder, ...newlyAddedPagesOrder];
      setDoc(newDoc);
      setPDF(pdf);
      setPagesOrder(newOrder);
```

If we do that I think that last step in the table above would be:
| upload 2 pages |   1 2 3 4 5 6 7  | 3 1 4 5 6 7  |

Reply:
Yes that's what i.e. actually the indexes are duplicated, i'll look into this more not sure if the solution you introduced solves the issue. because the ```newPagesIndexOffset``` will always be the number of pages in pdf because it's constant and not being changed.

for example a page of 5 the resulting ```newlyAddedPagesOrder``` would be something like
```[6, 7, 8, 9]``` right ? or am i missing something.

Reply:
> Because the newPagesIndexOffset will always be the number of pages in the pdf as it's constant and they are not being modified.

> for example a page of 5 the resulting newlyAddedPagesOrder would be something like
[6, 7, 8, 9] right ? or am i missing something.

Yes. Which is exactly what we want to fix this. We want to make sure the "order" we create refers to the proper pages. 

An alternative to all of this is we could still keep the "Deleted page" and not remove it from Order but instead just "mark it as "deleted" visually. Not sure if this is nice but it does enable a feature of "undoing" deleting any page that is currently shown to be deleted. It just might be confusing or might get too crowded if humans delete a lot of pages